doc: update HomeKit known issues (#10724)
nrfconnect/sdk-nrf: doc/nrf/known_issues.rst

HomeKit
=======

.. rst-class:: v2-3-0 v2-2-0

> 
> KRKNWK-16575: Apple DFU doesn't work on nRF5340
>   App core not upgraded due to problem with uploading images for two cores. Net core image upload override already uploaded app core image.
> 
>   **Workaround:** Manually cherry-pick and apply commit from the main branch (commit hash: ``TBD``).

Reviewer: ```suggestion

KRKNWK-16575: Apple DFU does not work on the nRF5340 DK
  Application core cannot be upgraded due to a problem with uploading images for two cores. 
  Uploading the net core image overrides an already uploaded application core image.

  **Workaround:** Manually cherry-pick and apply commit from the main branch (commit hash: ``TBD``).
```

HomeKit
=======

.. rst-class:: v2-3-0 v2-2-0

> KRKNWK-16503: OTA DFU using the iOS Home app (over UARP) does not work on nRF53 boards

Reviewer 2: "OTA DFU using the iOS Home app (over UARP)" is how we used to call this DFU solution in release notes. So suggested change affects the meaning.

> KRKNWK-16503: OTA DFU using the iOS Home app (over UARP) does not work on nRF53 boards

Reviewer: got it - updated the suggestion, thank you!
